django/forms/models.py

```diff
@@ -1112,7 +1112,7 @@ def __iter__(self):
             yield self.choice(obj)
 
     def __len__(self):
-        return (len(self.queryset) + (1 if self.field.empty_label is not None else 0))
+        return (len(self.queryset.all()) + (1 if self.field.empty_label is not None else 0))
```

Comment:
That's what I thought, I'm not sure why the test is unaffected. Anyway, I'll close the PR as I'm not investigating this is more detail right now.